src/agent/coverage/fuzz/fuzz_targets/fuzz_target_record_coverage.rs

```diff
@@ -0,0 +1,51 @@
+#![no_main]
+
+use libfuzzer_sys::fuzz_target;
+use std::env;
+use std::fs;
+use std::io::Write;
+use std::process::Command;
+use std::sync::Arc;
+use std::time::Duration;
+
+use tempfile::NamedTempFile;
+
+use coverage::allowlist::TargetAllowList;
+use coverage::binary::BinaryCoverage;
+use coverage::record::CoverageRecorder;
+
+use debuggable_module::loader::Loader;
+
+const INPUT_MARKER: &str = "@@";
+
+fuzz_target!(|data: &[u8]| {
+    if data.len() == 0 {
+        return;
+    }
+
+    // Write mutated bytes to a file
+    let mut file = NamedTempFile::new_in(env::current_dir().unwrap()).unwrap();
+    file.write_all(data);
+    let path = String::from(file.path().to_str().unwrap());
+
+    // Make sure the file is executable
+    Command::new("chmod").args(["+wrx", &path]).spawn().unwrap();
```

Comment:
This should be waited upon to finish:
```suggestion
    Command::new("chmod").args(["+wrx", &path]).spawn().unwrap().wait();
```

I think also that once this is done you don't need the `file.keep` or `fs::remove_file` calls; the file should be kept until the end of the function.

Reply:
The reason I had `file.keep`/`fs::remove_file` was that tempfile kept a handle open on the file (and deleting the file on `drop`) which was causing issues with our binary inspection.